syntax = "proto3";

import "google/protobuf/timestamp.proto";
import "foxglove/Vector3.proto";
import "foxglove/Quaternion.proto";

package keelson;

message ImuReading {
  google.protobuf.Timestamp timestamp = 1;

  string frame_id = 2;

  foxglove.Quaternion orientation = 3;
  repeated double orientation_covariance = 4; // array with 9 elements, row major

  foxglove.Vector3 angular_velocity = 5;
  repeated double angular_velocity_covariance = 6; // array with 9 elements, row major

  foxglove.Vector3 linear_acceleration = 7;
  repeated double linear_acceleration_covariance = 8; // array with 9 elements, row major
}
